src/patternfly/components/Nav/nav.scss

```diff
@@ -1010,6 +1022,10 @@
     --#{$nav}__subnav--PaddingLeft: var(--#{$nav}__subnav__subnav--PaddingLeft);
   }
 
+  > .#{$nav}__list {
+    --#{$nav}__list--PaddingTop: 0;
```

Comment:
This works, but I believe you can just set this under https://github.com/patternfly/patternfly/blob/4659b8c6202d1e5fe07cfe3093557ef30997f45e/src/patternfly/components/Nav/nav.scss#L995